app/image/image.go

```diff
@@ -303,6 +305,25 @@ func DeleteAllAppImageNames(appName string) error {
 	return coll.RemoveId(appName)
 }
 
+func UpdateAppImageRollback(appName, img, reason string, rollback bool) error {
+	dataColl, err := imageCustomDataColl()
+	if err != nil {
+		return err
+	}
+	defer dataColl.Close()
+	err = dataColl.Update(bson.M{"_id": fmt.Sprintf("%s:%s", appBasicImageName(appName), img)},
```

Comment:
The received image id here should already be a full image id (see previous comment on app.RollbackUpdate). So there's no need for this call to `fmt.Sprintf` simply filter with `{"_id": img}`.